lib/block_view_validator.go

```diff
@@ -1919,39 +1854,61 @@ func (bav *UtxoView) GetTopActiveValidatorsByStake(limit uint64) ([]*ValidatorEn
 	return validatorEntries[0:upperBound], nil
 }
 
-func (bav *UtxoView) GetGlobalActiveStakeAmountNanos() (*uint256.Int, error) {
-	// Read the GlobalActiveStakeAmountNanos from the UtxoView.
-	if bav.GlobalActiveStakeAmountNanos != nil {
-		return bav.GlobalActiveStakeAmountNanos.Clone(), nil
+func (bav *UtxoView) JailInactiveSnapshotValidators(blockHeight uint64) error {
+	// Check if we have switched from PoW to PoS yet. If we have not, then the PoS consensus
+	// has not started. We don't want to jail any validators until they have had the opportunity
+	// to participate in the consensus and are known to be inactive.
+	if blockHeight < uint64(bav.Params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight) {
+		return nil
 	}
-	// If not set, read the GlobalActiveStakeAmountNanos from the db.
-	globalActiveStakeAmountNanos, err := DBGetGlobalActiveStakeAmountNanos(bav.Handle, bav.Snapshot)
+
+	snapshotGlobalParams, err := bav.GetSnapshotGlobalParamsEntry()
 	if err != nil {
-		return nil, errors.Wrapf(err, "UtxoView.GetGlobalActiveStakeAmountNanos: ")
+		return errors.Wrapf(err, "UtxoView.JailInactiveSnapshotValidators: ")
```

Comment:
Let's add some more details to the errors in this function so it's easier to pinpoint where they came from if we hit them.